x/txfees/keeper/hooks.go

```diff
@@ -168,13 +161,12 @@ func (h Hooks) AfterPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId u
 		}
 
 		// get the swapRoute for the 2nd pool asset
-		var route []pooltypes.SwapAmountInRoute
-
-		feeToken, err := h.k.GetFeeToken(ctx, registeredDenom)
+		feeToken, err = h.k.GetFeeToken(ctx, registeredDenom)
 		if err != nil {
 			h.k.Logger(ctx).Error("failed to get fee token", "error", err)
 			return
 		}
+		var route []pooltypes.SwapAmountInRoute
```

Comment:
nit: we know the capacity in advance
```suggestion
		route := make([]pooltypes.SwapAmountInRoute, 0, len(feeToken.Route)+1)
```